### ISSUE TITLE
Sphinx docs haven't worked in over a year (for me)

### ISSUE THREAD

**Issue by jrmarino** [AdaCore/gprbuild#26]
I haven't issues with Sphinx documentation on other Adacore products, but for gprbuild its has never worked.  I just tried again to see if anything improved since a year ago, but alas, no.

```
Exception occurred:
  File "/raven/lib/python3.5/re.py", line 173, in search
    return _compile(pattern, flags).search(string)
TypeError: cannot use a string pattern on a bytes-like object
The full traceback has been saved in /tmp/sphinx-err-hh11rjhf.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
ignoring txt
ignoring gprbuild_gps.xml
ignoring Makefile
ignoring html
*** [txt] Error code 1
```

Just writing an issue in case this is unknown to developers.  Note that today I tried python 3.5, but it didn't work with python 2.7 last year either.  I just used the same 3.5 sphinx with xmlada, so I don't believe that's the culprit.

**Comment by t-14** [comment]
Even if it works for xmlada, given the crash location, it still does look like a bug in your sphinx distrib. FWIW we are using version 1.5 in house AFAIK, and the doc builds fine.

**Comment by jrmarino** [comment]
The version I used for this was 1.5.5.
Thus diagnosis is illogical.
Closing the ticket on the basis of version therefore is premature because obviously that's not the problem.
Closing tickets on potentially (actually likely) valid issues don't solve them, but it will discourage people from submitting new ones.  FWIW, this gprbuild sphinx issue persists along multiple sphinx versions on multiple operating systems despite sphinx docs working fine on every other adacore package.  All we know for sure is that it works on your specific setup in house.

**Comment by t-14** [comment]
> The version I used for this was 1.5.5.

You said you used "3.5 sphinx" actually. I realize now you probably meant "python" here. We are using 2.7 FWIW. Can you confirm that you see the same issue with 2.7?

> Closing the ticket on the basis of version therefore is premature

That's not why I closed it. The only rule I follow is: is there an action item for us? Answer AFAICS is no, because 1) we can't reproduce it, and 2) you failed to provide any details or context that would allow us to guess what's wrong.

I simply have no clue how you want us to investigate this. If you go half way by identifying what you think is wrong with our docs I will be happy to reopen it.

> All we know for sure is that it works on your specific setup in house.

That's one way of seeing it. Another way of seeing it is that all we know for sure is that it doesn't work on *your* specific setup. ;)

**Comment by jrmarino** [comment]
In the first post, I mentioned it hasn't worked with python 2.7 either.

As for open/close status:
I know I didn't provide that much information.  I didn't expect it to be fixed.  I did, however, expect the issue to stay open so that somebody else could see it was an issue and say "me too" or contribute to it.

It seems that every other project leaves issue open until they are confirmed to be bogus or after complete investigation it's decided to be a "wont fix" type of thing.

By closing valid issues, it sends the message that A) you care about low issue counts (e.g. no open issues means gprbuild is bug-free) and B) the little effort I spent to come to this site and document my observations were wasted.  No other people will see it because it's closed and C) it's not going to get fixed because it's considered bogus.

All I know is that I've seen it broken on more platforms than you've seen it work on.
My guess is that it is a locale thing, and something linux-specific allows it to work on linux only.  It's just a guess.

My advice is to leave real issues open even if you have no immediate action.  Actions might come later when additional information is known, but without a tracking PR to glue them altogether, that information won't come together.

**Comment by t-14** [comment]
> It seems that every other project leaves issue open until they are confirmed to be bogus or after complete investigation it's decided to be a "wont fix" type of thing.

What does "confirmed to be bogus" mean? I have no doubt it's not "bogus", i.e. you do see what you see. But I can't act on it because it works for me. Look at bugzilla workflow. https://wiki.documentfoundation.org/File:BzLifecycleold.png - in full accordance with the cycle we went through the stages UNCONFIRMED->ASSIGNED->RESOLVED:WORKSFORME->CLOSED.

As far as I am concerned, all that keeping open unactionable issues does is drown the actionable ones, i.e. it's harmful, not helpful.

> No other people will see it because it's closed

How would that affect their own experience? Either it works for them or it doesn't, and in the later case they'll report it. There is no need for "me-too's", we treat every issue we can, not just those that affect a greater number of  people than 1.

> All I know is that I've seen it broken on more platforms than you've seen it work on.

Hmm how did you arrive at this conclusion? For instance I am writing to you on my Mac laptop with a Windows VM and I verified that it works on both.

**Comment by jrmarino** [comment]
Issues and bug tracking is not a new thing.
Your approach of burying real issues to have a "clean table" is not conventional.
As I said, all you've done is inform me and anyone that may be monitoring that writing an issue is a waste of time unless we diagnose it ourselves and deliver the entire cause/effect/fix on a silver platter.

If you think the issue is real, closing this PR for "cleanliness" is conventionally wrong.
You're quoting a workflow but in my experienced opinion, you did not follow it.  (there's a few steps of evaluation you skipped.  You didn't resolve it, nor did you do any realistic diagnosis.    You noted that it's a WORKSFORME situation and closed it).

We're spinning our wheels.  If what I am saying is a foreign concept then your belief is likely deeply ingrained.  

Also, this is not "unactionable".  There are PLENTY of actions you could take.  I'm not forcing you to take any action, but to claim there's nothing that can be done is just incorrect.  Or in other words: if I put a $5000 bounty on diagnosing the problem and coming up with a fix, I'm sure I'd have to pay that bill within 48 hours.  

But lets just forget it.  I was trying to do a good deed and feel punished.  I honestly forgot this "immediate issue closure policy".  I'll try to remember it in the future to save us both time.

**Comment by t-14** [comment]
> unless we diagnose it ourselves and deliver the entire cause/effect/fix on a silver platter

Fix on a silver platter would sure be very welcome, if for no other reason than it would allow me to finally understand what is it that you wanted fixed in the first place. However implying that one is required to provide the fix by themselves is simply not fair, as the history of closed issues shows. Such rhetorical exaggerations,  talk of $5000 bounties, etc, is not helpful in any manner.

>  I was trying to do a good deed and feel punished. I honestly forgot this "immediate issue closure policy".

I am sorry you feel this way, but there is no such policy, otherwise there wouldn't be any issues still open. However, minimal investigation when we can't reproduce it is indeed expected from the submitter. If you are unwilling to undertake it, that's fine, but I see no point keeping such an issue open.

**Comment by jrmarino** [comment]
Wait, you don't even understand what the issue is about?  And you closed it anyway?
The problem is that gprbuild document generation has been broken on at least BSD platforms for over a year.  The impact is that there is no documentation provided in gprbuild packages.  That cascades to every user of GPRbuild on BSD platforms.  Every other adacore software has no problem generating documentation with sphinx;  it's specific to gprbuild.  As I packaged gprbuild on multiple package system (pkgsrc, freebsd ports) this affects many people and many platforms.

As for how I feel: I'm simply not impressed with your philosophy.  It's not a case of "to each his own".  Some approaches are better than others and the approach I'm seeing here is A) rare and B) not good.

I mentioned an exaggerated bounty to illustrate that it's a matter of motivation.  If there was an impossible to fix/diagnose bug, it wouldn't be resolved even if I offered 1 million dollars.  I was saying that I'm guessing this bug is not actually terrible difficult to either diagnose nor fix, so if $5000 would get it done, there are clearly actions that could (reasonably) be taken.

With each bug report, one lists the "impact of not fixing".
The impact here is that nobody gets gprbuild documentation with their package.  How severe is that?  I guess not terribly as long as they have a desktop machine with a browser connected to the internet.   Everyone affected has lived with it so far.  As a project owner, I would personally be motivated to at least investigate more than I think you've done, but that's just me.  I'm not going to push this anymore.
